src/lib/marquee.jsx

```diff
@@ -65,45 +80,52 @@ export const Marquee = ({
     }
 
     return marquees
-  }
+  }, [children, neededAmount])
 
   useEffect(() => {
-    getNumber()
+    setState()
+
+    const { current } = container
 
     const handleResize = () => {
-      if (resizeTimer.current) clearTimeout(resizeTimer.current)
+      if (current) clearTimeout(current)
+
       resizeTimer.current = setTimeout(() => {
-        getNumber()
+        setState()
       }, 200)
     }
 
     window.addEventListener("resize", handleResize)
 
     return () => {
+      clearTimeout(current)
       window.removeEventListener("resize", handleResize)
     }
-  }, [getNumber])
+  }, [setState])
 
   return (
     <div
+      {...props}
```

Comment:
I feel like spreading props can be a little unsafe, usage-wise. Is it alright with you if we remove this? I understand it's up to the developer not to f*** this up, but maybe we just make our component a bit more defensive.

If there were a need for any ARIA attributes (as an example), I feel like the dev could just wrap this in a parent div with 'em. Let me know if you envision a different use-case for the prop-spreading though.

Reply:
If it can be done with a single element, why force the dev to add a second node just to have some aria or whatever attributes they need?  

I strongly believe that a `div` from a custom component should support all attributes that a regular `div` does by default. If `forwardRef` wasn't so much hassle I'd have that as default as well to be honest.

Reply:
All good points. Thanks for the explanation @marlonmarcello